src/Nacara.Core/Menu.fs

```diff
@@ -37,3 +37,26 @@ let toFlatMenu (menu : Menu) =
         | MenuItem.List _ ->
             failwith "Should not happen because all the MenuItem.List should have been flattened"
     )
+
+/// <summary>
+/// It returns a MenuItem.List if it finds a page with the given pageId under a MenuItem.List.
+///
+/// Used for obtaining the section name for navigation buttons.
+/// </summary>
+/// <param name="menu">Menu where to look for a page ID.</param>
+/// <param name="pageId">ID of the page to look for.</param>
+/// <returns>MenuItem.List or MenuItem.Page or None.</returns>
+let rec tryFindSection (menu : Menu) pageId =
```

Comment:
The functions looks good to me.

It kind of assume we have only a 2 level deeps menu or want to print only the deepest section.

So if the page is inside of `Section 1 > Section 2 > My Page`, the button will only display `Section 2`.

This is not a problem as we agreed to have only 2 levels deep menu in #72 and task is tracked in #88 for me to do.

Reply:
If I am not mistaken it always returns the most top section. In your example it would return "Section 1" for "My Page".

Reply:
Ah ok, I just read quickly the code before taking my evening ^^

But, the comment still stand but there is nothing to do on your side. As the menu will always be 2 level deep in the future #88  so at max "Section > Page"